### PR TITLE
docs: add docs for serverInfo and connector functions in java client

### PR DESCRIPTION
**Pull request by jzaralim**
### Description 
Docs for the following java client functions: serverInfo, createConnector, dropConnector, listConnectors and describeConnector.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 1]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
 System.out.println(description.readQueries().size() + " queries read from this stream/table.");
 ``` 
 
+Get metadata about the ksqlDB cluster
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:696]: We should update the table of contents at the top of this file with the new sections so they're more discoverable.

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 2]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
...
+------------------------------------------------------------------
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:697]:
```suggestion
-------------------------------------
```

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 3]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
...
+Manage, list and describe connectors
+------------------------------------------------------------------
+
+Starting with ksqlDB 0.18.0, the Java client for ksqlDB supports the following connector operations:
+
+- Creating new connectors by using the `createConnector()` method
+- Dropping existing connectors by using the `dropConnector()` method
+- Listing connectors by using the `listConnectors()` method
+- Describing a specific connector by using the `describeConnector()` method
+
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:723]: Add a link to the API reference, either here or for the individual methods.

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 4]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
...
+```java
+Map<String, String> connectorProperties = ImmutableMap.of(
+  "connector.class", "io.confluent.connect.jdbc.JdbcSourceConnector",
+  "connection.url", "jdbc:postgresql://localhost:5432/my.db",
+  "mode","bulk",
+  "topic.prefix","jdbc-",
+  "table.whitelist","users",
+  "key","username"
+);
+client.createConnector("jdbc-connector", true, connectorProperties);
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:736]:
```suggestion
client.createConnector("jdbc-connector", true, connectorProperties).get();
```

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 5]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
...
+- Describing a specific connector by using the `describeConnector()` method
+
+### Example Usage ###
+
+Create a new connector:
+```java
+Map<String, String> connectorProperties = ImmutableMap.of(
+  "connector.class", "io.confluent.connect.jdbc.JdbcSourceConnector",
+  "connection.url", "jdbc:postgresql://localhost:5432/my.db",
+  "mode","bulk",
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:731]: nit: inconsistent spacing in this example -- add a space on this line and the ones below?

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 6]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
...
+  "topic.prefix","jdbc-",
+  "table.whitelist","users",
+  "key","username"
+);
+client.createConnector("jdbc-connector", true, connectorProperties);
+```
+
+Drop a connector:
+```java
+client.dropConnector("jdbc-connector");
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:741]:
```suggestion
client.dropConnector("jdbc-connector").get();
```

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 7]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
...
+```
+
+List connectors:
+```java
+List<ConnectorInfo> connectors = client.listConnectors();
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:746]:
```suggestion
List<ConnectorInfo> connectors = client.listConnectors().get();
```

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 8]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
...
+for (ConnectorInfo connector : connectors) {
+  System.out.println(connector.name()
+    + " " + connector.type()
+    + " " + connector.className()
+    + " " + connector.state()
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:751]: What are possible values returned by this? We don't need to add it here but we should update the javadoc for this method to include this information.

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 9]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
...
+  );
+}
+```
+
+Describe a connector:
+```java
+ConnectorDescription description = client.describeConnector("jdbc-connector");
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:758]:
```suggestion
ConnectorDescription description = client.describeConnector("jdbc-connector").get();
```

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 10]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
...
+System.out.println(description.name()
+  + " is a " + description.type() + " connector."
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:760]: nit: add newline after the periods in this output.

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 11]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
...
+Fetch server metadata:
+```java
+ServerInfo serverInfo = client.serverInfo().get();
+System.out.println("The ksqlDB version running on this server is " + serverInfo.getServerVersion());
+System.out.println("The Kafka cluster this server is using is " + serverInfo.getKafkaClusterId());
+System.out.println("The id of this ksqlDB service is " + serverInfo.getKsqlServiceId());
+``` 
+
+Manage, list and describe connectors
+------------------------------------------------------------------
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:715]:
```suggestion
------------------------------------
```

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 12]
```diff
@@ -691,6 +693,78 @@ System.out.println(description.writeQueries().size() + " queries write to this s
 System.out.println(description.readQueries().size() + " queries read from this stream/table.");
 ``` 
 
+Get metadata about the ksqlDB cluster
+------------------------------------------------------------------
+
+Starting with ksqlDB 0.16.0, the `serverInfo()` method enables client apps to fetch metadata about
```
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:699]: Ideally this piece would be in a separate PR so we can get the change live for 0.16 ahead of the connector docs, but it's not a huge deal; your call on whether you think it's worth the effort.

**Review by vcrfxia — COMMENTED** [review]
Thanks @jzaralim -- bunch of comments inline but LGTM overall!

File: docs/developer-guide/ksqldb-clients/java-client.md (hunk 12 above)
Comment (JimGalasyn) [reply docs/developer-guide/ksqldb-clients/java-client.md:699]: Joel and I had a chat with MD and Apurva, and we're trying to get me out of the Mechanical Turk business. :)

**Review by JimGalasyn — APPROVED** [review]
LGTM, modulo suggestions by @vcrfxia.

**Review by vcrfxia — APPROVED** [review]
🎉 Thanks @jzaralim !

File: docs/developer-guide/ksqldb-clients/java-client.md (hunk 12 above)
Comment (jzaralim) [reply docs/developer-guide/ksqldb-clients/java-client.md]: Since the rest of the docs will be coming out for 0.18, I'll move this to another PR.

**Pull request merged**